doc/src/manual/methods.md

```diff
@@ -335,7 +336,20 @@ Here's an example where the method type parameter `T` is used as the type parame
 type `Vector{T}` in the method signature:
 
 ```jldoctest
-julia> myappend(v::Vector{T}, x::T) where {T} = [v..., x]
+julia> function myappend(v::Vector{T}, x::T) where {T}
+           return [v..., x]
+       end
+myappend (generic function with 1 method)
+```
+
+The type parameter `T` in the function's signature ensures that the type of the new `v` is preserved after element `x` is appended to it. 
```

Comment:
```suggestion
Due to the type parameter `T` in the function's signature, this method only applies if `x` is of the element type of `v`.
```

Talking about a “new `v`” is misleading, since the original vector is not changed. There is only a “new `v`” if you call the function like `v = myappend(v, x)`. If you call it like `something_else = myappend(v, x)`, `v` still has the old value.

Reply:
Ah yes. How about this? 

The type parameter `T` in the function's signature constrains the method to operate on the same types: element `x` is appended to `v` only if it is of the same type as the elements of `v`, otherwise a method error is returned. 

(Am I understanding correctly? Thanks.)

Reply:
To me, “constrains the method to operate on the same types” sounds wrong, since `v` and `x` are not of the same type (`v` is a `Vector`, `x` not necessarily). Also, the `MethodError` is thrown, not returned.

Reply:
How about this:

The type parameter `T` in the method signature restricts the possible argument types: the method applies only when `x` is of the same type as the elements of `v`.

Reply:
Yes, that's clear to me. Do we also append "and throws a method error otherwise" ?

Reply:
I think it would be best to use the sentence from below the second code block before this one, i.e.:

> As you can see, the type of the appended element must match the element type of the vector it is appended to, or else a `MethodError` is raised. The type parameter `T` in the method signature restricts the possible argument types: the method applies only when `x` is of the same type as the elements of `v`.

The sentence could then be removed from the second code block. I think it is better to keep the explanation in one paragraph, not part of it below the first and part below the second code block. To make this consistent, we should also move the examples into the first code block, so that the second code block only shows the short-form syntax.